gemma-core/src/main/java/ubic/gemma/core/datastructure/matrix/ExpressionDataMatrix.java

```diff
@@ -220,6 +220,23 @@
      */
     T[][] getRows( List<CompositeSequence> designElements );
 
+    /**
+     * Access a submatrix
+     * @param rowIndices of integers to select
+     * @return T[][] or null if rowIndices is null or empty.
+     */
+    T[][] getRows( Collection<Integer> rowIndices);
```

Comment:
How about just a `getRowsByName` and avoid altogether constructing a collection of integers to collect relevant rows?

Reply:
> How about just a `getRowsByName` and avoid altogether constructing a collection of integers to collect relevant rows?

I prefer it this way (getRows) to keep the semantics of the named data matrix. The "name" for the row is an object in this case a CompositeSequence. For most operations the innards of that are not exposed (except when we serialize it). getRowsByName would take a collection of CompositeSequences as the parameter.

The method I added is primarily for debugging purposes, so I can test if an expected row is there, and I don't want to deal with the database, and I know the probe name.

Reply:
It's good, I see the value in having a method for selecting specific rows by indices.